tests-js/Utils/initializer.js

```diff
@@ -0,0 +1,358 @@
+};
+
+const open_margin_trade_position = async (
+	loanToken,
+	RBTC,
+	WRBTC,
+	SUSD,
+	trader,
+	collateral = "RBTC",
+	loan_token_sent = hunEth.toString(),
+	leverage_amount = new BN(2).mul(oneEth).toString()
+) => {
+	await SUSD.mint(trader, loan_token_sent);
+	await SUSD.approve(loanToken.address, loan_token_sent, { from: trader });
+
+	let collateralToken;
+	if (collateral == "RBTC") collateralToken = RBTC.address;
+	else collateralToken = WRBTC.address;
+
+	const { receipt } = await loanToken.marginTrade(
+		"0x0", // loanId  (0 for new loans)
+		leverage_amount, // leverageAmount
+		loan_token_sent, // loanTokenSent
+		0, // no collateral token sent
+		collateralToken, // collateralTokenAddress
+		trader, // trader,
+		[], // loanDataBytes (only required with ether)
+		{ from: trader }
+	);
+	const decode = decodeLogs(receipt.rawLogs, LoanOpenings, "Trade");
+	return [decode[0].args["loanId"], trader, loan_token_sent, leverage_amount];
+};
+
+const open_margin_trade_position_iBTC = async (
+	loanTokenWRBTC,
+	SUSD,
+	trader,
+	loan_token_sent = oneEth.toString(),
+	leverage_amount = new BN(2).mul(oneEth).toString()
+) => {
+	const { receipt } = await loanTokenWRBTC.marginTrade(
+		"0x0", // loanId  (0 for new loans)
+		leverage_amount, // leverageAmount
+		loan_token_sent, // loanTokenSent
+		0, // no collateral token sent
+		SUSD.address, // collateralTokenAddress
+		trader, // trader,
+		[], // loanDataBytes (only required with ether)
+		{ from: trader, value: loan_token_sent }
+	);
+
+	const decode = decodeLogs(receipt.rawLogs, LoanOpenings, "Trade");
+	return [decode[0].args["loanId"], trader, loan_token_sent, leverage_amount];
+};
+
+const borrow_indefinite_loan = async (
+	loanToken,
+	sovryn,
+	SUSD,
+	RBTC,
+	accounts,
+	withdraw_amount = oneEth.toString(),
+	margin = new BN(50).mul(oneEth).toString(),
+	duration_in_seconds = 60 * 60 * 24 * 10
+) => {
+	const borrower = accounts[2];
+	const receiver = accounts[1];
+	const collateral_token_sent = await sovryn.getRequiredCollateral(SUSD.address, RBTC.address, withdraw_amount, margin, true);
+	// approve the transfer of the collateral
+	await RBTC.mint(borrower, collateral_token_sent);
+	await RBTC.approve(loanToken.address, collateral_token_sent, { from: borrower });
+	// borrow some funds
+	const tx = await loanToken.borrow(
+		constants.ZERO_BYTES32, // bytes32 loanId
+		withdraw_amount, // uint256 withdrawAmount
+		duration_in_seconds, // uint256 initialLoanDuration
+		collateral_token_sent, // uint256 collateralTokenSent
+		RBTC.address, // address collateralTokenAddress
+		borrower, // address borrower
+		receiver, // address receiver
+		"0x", // bytes memory loanDataBytes
+		{ from: borrower }
+	);
+	const decode = decodeLogs(tx.receipt.rawLogs, LoanOpenings, "Borrow");
+	const loan_id = decode[0].args["loanId"];
+	return [loan_id, borrower, receiver, withdraw_amount, duration_in_seconds, margin, tx];
+};
+
+function decodeLogs(logs, emitter, eventName) {
```

Comment:
Yes it can be used for comparing equal values from event to expected. 
But ExpectEvent.InTransaction is not relevant for checking differences between real value and expected value
like realResult > expectedResult or realResult < expectedResult